neqo-transport/src/connection/mod.rs

```diff
@@ -1142,6 +1153,25 @@ impl Connection {
 
         match (&self.state, self.role) {
             (State::Init, Role::Client) => {
+                // Before starting the client, send some noise
+                if self.conn_params.pre_init_noise_enabled() && self.pre_initial_pkts > 0 {
```

Comment:
It should be enough to just check the counter.  The counter should be zero otherwise.